### PR TITLE
SCB-865 Implement reaction of the event in Alpha Server

### PR DESCRIPTION
**Pull request by cherrylzhao**
Implement reaction of TCC event in alpha server, which involves
-  rpc OnConnected (GrpcServiceConfig) returns (stream GrpcTccCordinateCommand)
- rpc participate(GrpcTccParticipateEvent) returns (GrpcAck)
- rpc OnTccTransactionEnded (GrpcTccTransactionEndedEvent) returns (GrpcAck)
- rpc OnDisconnected (GrpcServiceConfig) returns (GrpcAck)
- add junit test framework code for alpha TCC workflow.

### CODE REVIEW

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/GrpcOmegaTccCallback.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import io.grpc.stub.StreamObserver;
+import org.apache.servicecomb.saga.alpha.tcc.server.event.ParticipatedEvent;
+import org.apache.servicecomb.saga.common.TransactionStatus;
+import org.apache.servicecomb.saga.pack.contract.grpc.GrpcTccCoordinateCommand;
+
+/**
+ * Grpc omega callback for TCC workflow.
+ *
+ * @author zhaojun
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/GrpcOmegaTccCallback.java]: We don't use the author annotation in our code, people can look it up by using git .

File: alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcBootstrap.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.Properties;
+import javax.net.ssl.SSLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GrpcBootstrap implements Bootstrap {
```
Comment (WillemJiang) [inline alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcBootstrap.java]: Can we still use the Bootstrap from the saga server?

File: alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcTccServerConfig.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ *  limitations under the License.
+ */
+
+package org.apache.servicecomb.saga.alpha.tcc.server.common;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class GrpcTccServerConfig {
```
Comment (WillemJiang) [inline alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcTccServerConfig.java]: we should share the configuration of saga alpha server.

File: alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/TccCoordinateCommandStreamObserver.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Override
+  public void onNext(GrpcTccCoordinateCommand value) {
+    consumer.accept(value);
+    receivedCommands.add(value);
+  }
+
+  @Override
+  public void onError(Throwable t) {
+
```
Comment (WillemJiang) [inline alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/TccCoordinateCommandStreamObserver.java]: The error means there is something wrong when sending the CoordinateCommand back to Omega. what's the purpose of receivedCommands?

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/event/ParticipatedEvent.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+package org.apache.servicecomb.saga.alpha.tcc.server.event;
+
+import org.apache.servicecomb.saga.common.TransactionStatus;
+
+/**
+ * Participate event.
+ *
+ * @author zhaojun
+ */
+public class ParticipatedEvent {
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/event/ParticipatedEvent.java]: I don't think we need this event in  alpha server, as we can use the grpc generated Event directly.

**Comment by WillemJiang** [comment]
We need update the unit test, as the code coverage is dropped a lot.

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/GrpcOmegaTccCallback.java (hunk 1 above)
Comment (cherrylzhao) [reply alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/GrpcOmegaTccCallback.java]: ok

File: alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcBootstrap.java (hunk 2 above)
Comment (cherrylzhao) [reply alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcBootstrap.java]: I can't reference it for saga server class (GrpcStartable) is not public,
so I create a GrpcBootstrap in test package

Comment (cherrylzhao) [reply alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/GrpcBootstrap.java]: now I don't want update the saga server code.

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/event/ParticipatedEvent.java (hunk 5 above)
Comment (cherrylzhao) [reply alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/tcc/server/event/ParticipatedEvent.java]: GrpcParticipatedEvent is probuf message, it is not convenient for us to handle backend process （eg. data persistence (reds or database)

File: alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/TccCoordinateCommandStreamObserver.java (hunk 4 above)
Comment (cherrylzhao) [reply alpha/alpha-server/src/test/java/org/apache/servicecomb/saga/alpha/tcc/server/common/TccCoordinateCommandStreamObserver.java]: this is Junit test class, receivedCommands simulate omega receive the coordinateCommand.

**Pull request closed without merging**